### PR TITLE
Added bzl_libraries for stardoc support

### PR DESCRIPTION
**Pull request by UebelAndre**
There's now a top level [bzl_library](https://github.com/bazelbuild/bazel-skylib#bzl_library) `@rules_foreign_cc//:bzl_srcs` that contain the `.bzl` sources for the build rules defined in this repo.

Stardoc support was also added to ensure the `bzl_library` structure is correct and usable outside of the repo. In addition to this, new documentation can be generated by running `bazel run //docs:generate_docs` from the root of `rules_foreign_cc/examples` to output `@rules_foreign_cc_tests//docs:README.md`.

### CODE REVIEW

**Comment by UebelAndre** [comment]
@oquenchil Hey, are you around to take a look at this? 🙏

**Comment by whs-dot-hk** [comment]
Hi, @UebelAndre I think this is unrelated to building (cmake, make, configure make, etc)

**Comment by UebelAndre** [comment]
This exposes the `.bzl` files as a `bzl_library` so that other rules/macros that depend on them can continue to have documentation generated via stardoc.

I ran into this because we replaced a custom rule with `configure_make` and broke our ability to generate documentation. If this PR gets merged, we no longer have to make the decision between generated documentation and a more canonical way to integrate older code into Bazel.

**Comment by UebelAndre** [comment]
@whs-dot-hk If you're interested, https://github.com/UebelAndre/rules_foreign_cc/tree/stardoc/examples/docs is the direct result of the changes in this PR (not including fixing CI) 😄 

I'd greatly appreciate having this merged. Documentation is really important to my team and one of our favorite parts of Bazel 😄

**Comment by UebelAndre** [comment]
I have another branch with all the docs plugged in. I find this incredibly helpful: [@rules_foreign_cc//examples/docs/...](https://github.com/UebelAndre/rules_foreign_cc/tree/a76e06d7603966233d1e3ef5696802e9f37c38be/examples/docs)

**Comment by UebelAndre** [comment]
@irengrig Hey, are you available to take a look at this PR? 🙏 😅

File: workspace_definitions.bzl [hunk 1]
```diff
@@ -53,37 +53,36 @@ _read_build_options = repository_rule(
...
-        additonal_shell_toolchain_package: a package under which additional toolchains, referencing
-        the generated data for the passed additonal_shell_toolchain_mappings, will be defined.
-        This value is needed since register_toolchains() is called for these toolchains.
-        Please refer to example in @rules_foreign_cc//toolchain_examples.
-    """
+        additonal_shell_toolchain_package: A package under which additional toolchains, referencing
+            the generated data for the passed additonal_shell_toolchain_mappings, will be defined.
+            This value is needed since register_toolchains() is called for these toolchains.
+            Please refer to example in @rules_foreign_cc//toolchain_examples.
+    """
```
Comment (jsharpe) [inline workspace_definitions.bzl:85]: nit: remove the trailing whitespace

**Review by jsharpe — CHANGES_REQUESTED** [review]
Looks good; its useful to have the functions documented properly without having to read through the code.

Just a couple of nits with formatting and an unrelated change to the top level WORKSPACE to remove and then this will be good to go.

It's probably worth adding a link to the docs on the top level README file too?

File: WORKSPACE [hunk 2]
```diff
@@ -2,17 +2,16 @@ workspace(name = "rules_foreign_cc")
 
 load("//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")
 
-rules_foreign_cc_dependencies([
-    "@rules_foreign_cc_tests//:built_cmake_toolchain",
-    "@rules_foreign_cc_tests//:built_ninja_toolchain_osx",
-    "@rules_foreign_cc_tests//:built_ninja_toolchain_linux",
-])
-
```
[the remark this answers is not in the document]
Comment (UebelAndre) [reply WORKSPACE:10]: Can do, it's been removed from this PR

[the remark this answers is not in the document]
Comment (UebelAndre) [reply WORKSPACE]: oops, forgot to push this change. Now it's been dropped.

**Review by jsharpe — APPROVED** [review]
LGTM

**Pull request merged**